src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -58,26 +58,18 @@ public static int Main(string[] args)
             AgentId = new Guid(args[0]);
             AgencyUrl = args[1];
             
-#if DEBUG
-            bool pause = false;
-#endif
             bool verbose = false;
             for (int i = 2; i < args.Length; i++)
                 switch (args[i])
                 {
-#if DEBUG
-                    case "--pause":
-                        pause = true;
+                    case "--debug-agent":
+                        if (!System.Diagnostics.Debugger.IsAttached)
+                            System.Diagnostics.Debugger.Launch();
```

Comment:
When I experimented with `Debugger.Launch()`, if there was a breakpoint set, execution would continue until the breakpoint was hit, but if no breakpoints were set, it would break on the next line. Is this behaviour not consistent? Does it depend on the running framework or on the debugger?